simple_salesforce/api.py

```diff
@@ -586,30 +604,16 @@ def _exception_handler(result, name=""):
     except Exception:
         response_content = result.text
 
-    if result.status_code == 300:
-        message = u"More than one record for {url}. Response content: {content}"
-        message = message.format(url=url, content=response_content)
-        raise SalesforceMoreThanOneRecord(message)
-    elif result.status_code == 400:
-        message = u"Malformed request {url}. Response content: {content}"
-        message = message.format(url=url, content=response_content)
-        raise SalesforceMalformedRequest(message)
-    elif result.status_code == 401:
-        message = u"Expired session for {url}. Response content: {content}"
-        message = message.format(url=url, content=response_content)
-        raise SalesforceExpiredSession(message)
-    elif result.status_code == 403:
-        message = u"Request refused for {url}. Response content: {content}"
-        message = message.format(url=url, content=response_content)
-        raise SalesforceRefusedRequest(message)
-    elif result.status_code == 404:
-        message = u'Resource {name} Not Found. Response content: {content}'
-        message = message.format(name=name, content=response_content)
-        raise SalesforceResourceNotFound(message)
-    else:
-        message = u'Error Code {status}. Response content: {content}'
-        message = message.format(status=result.status_code, content=response_content)
-        raise SalesforceGeneralError(message)
+    exc_map = {
+        300: SalesforceMoreThanOneRecord,
+        400: SalesforceMalformedRequest,
+        401: SalesforceExpiredSession,
+        403: SalesforceRefusedRequest,
+        404: SalesforceResourceNotFound,
+    }
+    exc_cls = exc_map.get(result.status_code, SalesforceGeneralError)
+
+    raise exc_cls(url, result.status_code, name, response_content)
```

Comment:
You could probably also pass in `result.url` as the first argument and remove the `url = result.url` higher up.